### PR TITLE
Allow karmabot to store notes for users, both in private and public

### PR DESCRIPTION
**Pull request by pmayd**
I added a new private and public command `note` with three subcommands:
- `note add <message>` stores a new note in the database with id, user_id as fk, timestamp and note message
- `note list` lists all notes from the db for the current user in the format <id> note from <timestamp>: <message>, with linebreak between each note
- `note del <id>` deletes note with <id> from database.

Added some tests and tried to be as failsave as possible.

Screenshot_2

### CODE REVIEW

**Review by pogross — APPROVED** [review]
ref #54  👍 @pmayd

File: src/karmabot/commands/note.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    """List all notes for a given user."""
+    notes = _get_notes_for_user(user)
+
+    if not notes:
+        return (
+            f"Sorry {user.username}, you don't have any notes so far! "
+            f"Just start adding notes via the 'note add' command."
+        )
+
+    msg = "\n".join(f"{i+1}. note {str(note)}" for i, note in enumerate(notes))
```
Comment (bbelderbos) [inline src/karmabot/commands/note.py:84]: No need to change, but could do `enumerate(notes, 1)`

**Comment by bbelderbos** [comment]
Awesome, thanks @pmayd looking forward using this feature soon.

File: src/karmabot/commands/note.py (hunk 1 above)
Comment (pmayd) [reply src/karmabot/commands/note.py:84]: ah thanks I always forget about that feature of enumerate!

**Pull request merged**